interchaintest/docker.go

```diff
@@ -29,7 +30,15 @@ type dockerErrorDetail struct {
 	Message string `json:"message"`
 }
 
-func BuildRelayerImage(t *testing.T) {
+func UniqueRelayerImageName() string {
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		fmt.Printf("Failed to generate UUID: %v\n", err)
+	}
+	return RelayerImagePrefix + uuid.String()[:6]
+}
+
+func BuildRelayerImage(t *testing.T, uniquestr string) {
```

Comment:
we should do the random image name generation within `BuildRelayerImage` and also add the `t.Cleanup` call to delete the image, that way it does not need to be repeated across tests and will automatically delete the image when the test ends, pass or fail.